src/components/MessageActions/hooks/useMessageActionsBoxPopper.ts

```diff
@@ -0,0 +1,45 @@
+import { Placement } from '@popperjs/core';
+import { useEffect, useRef } from 'react';
+import { usePopper } from 'react-popper';
+
+export interface MessageActionsBoxPopperOptions {
+  open: boolean;
+  placement: Placement;
+  referenceElement: HTMLElement | null;
+}
+
+export function useMessageActionsBoxPopper<T extends HTMLElement>({
+  open,
+  placement,
+  referenceElement,
+}: MessageActionsBoxPopperOptions) {
+  const popperElementRef = useRef<T>(null);
+  const { attributes, styles, update } = usePopper(referenceElement, popperElementRef.current, {
+    modifiers: [
+      {
+        name: 'eventListeners',
+        options: {
+          // It's not safe to update popper position on resize and scroll, since popper's
+          // reference element might not be visible at the time.
+          resize: false,
+          scroll: false,
+        },
+      },
+    ],
+    placement,
+  });
+
+  useEffect(() => {
+    if (open) {
+      // Since the popper's reference element might not be (and usually is not) visible
+      // all the time, it's safer to force popper update before showing it.
+      update?.();
+    }
+  }, [open]);
```

Comment:
Shouldn't we have `update` in the deps array as well?
```suggestion
  }, [open, update]);
```

Reply:
@oliverlaz `usePopper` hook keeps a stable instance of the popper object, so all methods have stable references as well - but I agree we should add `update` to the deps anyway!

Reply:
@oliverlaz Fixed in https://github.com/GetStream/stream-chat-react/pull/2241/commits/b5ce2301630dd123bff4d137ea6244276e06dadd.